lib/cartopy/mpl/gridliner.py

```diff
@@ -411,16 +578,48 @@ def _find_midpoints(self, lim, ticks):
                      self._round(np.percentile(lim, uq), cent))
         return midpoints
 
+    def _draw_this_label(self, xylabel, loc):
+        """Should I draw this kind of label here?"""
+        draw_labels = getattr(self, loc+'_labels')
+        if not draw_labels:
+            return False
+        if (isinstance(draw_labels, str) and draw_labels != xylabel):
+            return False
+        if ((xylabel == "x" and loc in ["left", "right"]) or
```

Comment:
I'm not sure there. There is not doubt now that x labels must not be drawn on lateral sides by default, and the reverse for y labels. 
However shouldn't we let the user finally decide. I mean, if the user explicitly says for instance `draw_labels={`top': 'y']}` or `draw_labels={`top': ['x', 'y']]}` then y labels are drawn on the top.
And if says `draw_labels={`top': True]}` then only the x labels are drawn.
What is your point of view ?

Reply:
I agree with you. The defaults are sensible, but if someone wants to override them then they should be allowed to do that.